### PR TITLE
Update links in the documentation

### PR DESCRIPTION
**Pull request by rzadp**
- Where applicable, use a regular [`reference`] instead of `../../../reference/index.html`.
- Typos.
- Update a link to `polkadot-evm` which has moved out of the monorepo.
- ~~The link specification for `chain_spec_builder` is invalid~~ (actually it was valid) - it works fine without it.

Part of https://github.com/paritytech/eng-automation/issues/10

### CODE REVIEW

File: docs/sdk/src/guides/your_first_pallet/mod.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@
 //! this guide, namely the crate/package names, based on which template you use.
 //!
 //! > Be aware that you can read the entire source code backing this tutorial by clicking on the
-//! > [`source`](./mod.rs.html) button at the top right of the page.
+//! > `source` button at the top right of the page.
```
Comment (rzadp) [inline docs/sdk/src/guides/your_first_pallet/mod.rs:24]: This link didn't work.
It is however possible to make it work - it would look like this:

```
[`source`](../../../src/polkadot_sdk_docs/guides/your_first_pallet/mod.rs.html)
```

(Unless I'm missing some other obvious way of linking to source)
I thought it's better to not have a link, since the user is pointed to the built-in button anyway.

File: docs/sdk/src/guides/your_first_pallet/mod.rs [hunk 2]
```diff
@@ -319,11 +319,11 @@
 //! - Learn more about the individual pallet items/macros, such as event and errors and call, in
 //!   [`frame::pallet_macros`].
 //!
-//! [`pallet::storage`]: ../../../frame_support/pallet_macros/attr.config.html
+//! [`pallet::storage`]: ../../../frame_support/pallet_macros/attr.storage.html
 //! [`pallet::call`]: ../../../frame_support/pallet_macros/attr.call.html
 //! [`pallet::event`]: ../../../frame_support/pallet_macros/attr.event.html
```
Comment (rzadp) [inline docs/sdk/src/guides/your_first_pallet/mod.rs:324]: In here, I think the proper way would be to link to `frame_support::pallet_macros::event`, instead of using `../../../`.

However the link then resolves a little differently (straight to `frame-support-procedural`) instead of `frame-support`, so in this case `../../../` type of linking is preferable.

File: docs/sdk/src/reference_docs/chain_spec_genesis.rs [hunk 3]
```diff
@@ -178,7 +178,6 @@
 //! [`pallet::genesis_build`]: frame_support::pallet_macros::genesis_build
 //! [`pallet::genesis_config`]: frame_support::pallet_macros::genesis_config
 //! [`BuildGenesisConfig`]: frame_support::traits::BuildGenesisConfig
-//! [`chain_spec_builder`]: ../../../staging_chain_spec_builder/index.html
```
Comment (ggwpez) [inline docs/sdk/src/reference_docs/chain_spec_genesis.rs:181]: The link is still used in the document a few times. It was moved to the [runtimes](https://github.com/polkadot-fellows/runtimes/tree/main/chain-spec-generator) repo, but not sure if it fits the context here.

Comment (rzadp) [reply docs/sdk/src/reference_docs/chain_spec_genesis.rs:181]: > The link is still used in the document a few times.

Yes, and the link continues to work because `chain-spec-builder` is [in Cargo.toml](https://github.com/paritytech/polkadot-sdk/blob/c8cb27fbd4920c8369aca053cb92186394d2199d/docs/sdk/Cargo.toml#L37).

> It was moved to the [runtimes](https://github.com/polkadot-fellows/runtimes/tree/main/chain-spec-generator) repo, but not sure if it fits the context here.

I can see `chain-spec-builder` still [inside polkadot-sdk](https://github.com/paritytech/polkadot-sdk/tree/c8cb27fbd4920c8369aca053cb92186394d2199d/substrate/bin/utils/chain-spec-builder) - it's not exactly the same thing as `chain-spec-generator` in `runtimes`, right?

File: docs/sdk/src/guides/your_first_pallet/mod.rs (hunk 2 above)
Comment (kianenigma) [reply docs/sdk/src/guides/your_first_pallet/mod.rs:324]: This should be fixed with https://github.com/paritytech/polkadot-sdk/pull/5023, no?

File: docs/sdk/src/guides/your_first_pallet/mod.rs (hunk 1 above)
Comment (kianenigma) [reply docs/sdk/src/guides/your_first_pallet/mod.rs:24]: I am surprised, why didn't the original one, or sth like `../mod.rs` work? I don't see why we have to back up 3 steps and re-navigate ourselves into the same directory?

File: docs/sdk/src/reference_docs/chain_spec_genesis.rs (hunk 3 above)
Comment (ggwpez) [reply docs/sdk/src/reference_docs/chain_spec_genesis.rs:181]: Oh yes 🙈  i mixed them up.

File: docs/sdk/src/guides/your_first_pallet/mod.rs (hunk 1 above)
Comment (rzadp) [reply docs/sdk/src/guides/your_first_pallet/mod.rs:24]: > back up 3 steps and re-navigate ourselves into the same directory

Not the same directory.

It navigates from:
`target/doc/polkadot_sdk_docs/guides/your_first_pallet/index.html`
to:
`target/doc/src/polkadot_sdk_docs/guides/your_first_pallet/mod.rs.html`

(note added `src`)

So it will link from [here](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/guides/your_first_pallet/index.html) to [here](https://paritytech.github.io/polkadot-sdk/master/src/polkadot_sdk_docs/guides/your_first_pallet/mod.rs.html).

File: docs/sdk/src/guides/your_first_pallet/mod.rs (hunk 2 above)
Comment (kianenigma) [reply docs/sdk/src/guides/your_first_pallet/mod.rs:324]: I see, so to recap, with the PR above, `frame_support_procedural` is the right place for the docs, but in that PR @gupnik didn't move all of them, and instead resided to a follow-up, but he is on holidays now :) 

TLDR: 

1. if you jump to `frame_support_procedural`, that's fine
2. in the meantime, you can try and follow in nikhil's footsptes and perhaps move a few more macros over? :) up to you

Comment (rzadp) [reply docs/sdk/src/guides/your_first_pallet/mod.rs]: Ok, understood.

So I switched to the proper links which jump to `frame_support_procedural`, which eventually will contain the docs.

I'll see if I can help with `2.` (separately - so the PR is ready for review).

File: docs/sdk/src/guides/your_first_pallet/mod.rs (hunk 1 above)
Comment (kianenigma) [reply docs/sdk/src/guides/your_first_pallet/mod.rs:24]: Okay, we can remove the link for now. 

As a wishlist item, I wish we could amend each page to have a few fixed buttons/links in a more pronounced way: 

- how to contribute to this page
- how to see the source of this page

The latter is already visible via the default rust HTML indeed.

**Review by kianenigma — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Review by mordamax — APPROVED** [review]

**Comment by ggwpez** [comment]
Marking this as `R0-silent` will not trigger a release of the crate and only include it in the next stable release.

**Comment by rzadp** [comment]
> Marking this as `R0-silent` will not trigger a release of the crate and only include it in the next stable release.

Hmm, I see that the crate is marked with `publish = false` and seems to stay on `0.0.1`:

https://github.com/paritytech/polkadot-sdk/blob/149c70938f2b29f8d92ba1cc952aeb63d4084e27/docs/sdk/Cargo.toml#L9-L11

So it seems to me like it's released a bit differently, do we still should treat it with a prdoc?

**Comment by rzadp** [comment]
Note: I [updated](https://github.com/paritytech/polkadot-sdk/pull/5175/commits/808e7f366b708b6cec9a203e8edb01d2c6509306) the favicon link which went missing with the new website release.
